examples/apps/ai-collab/src/components/TaskGroup.tsx

```diff
@@ -52,14 +54,116 @@ export function TaskGroup(props: {
 	const [isAiTaskRunning, setIsAiTaskRunning] = useState<boolean>(false);
 	const [llmBranchData, setLlmBranchData] = useState<{
 		differences: Difference[];
-		originalBranch: BranchableTree;
-		forkBranch: TreeBranchFork;
-		forkView: TreeView<typeof SharedTreeAppState>;
+		originalBranch: TreeViewAlpha<typeof SharedTreeAppState>;
+		forkBranch: TreeViewAlpha<typeof SharedTreeAppState>;
 		newBranchTargetNode: SharedTreeTaskGroup;
 	}>();
 
 	useSharedTreeRerender({ sharedTreeNode: props.sharedTreeTaskGroup, logId: "TaskGroup" });
 
+	/**
+	 * Helper function for ai collaboration which creates a new branch from the current {@link SharedTreeAppState}
+	 * as well as find the matching {@link SharedTreeTaskGroup} intended to be worked on within the new branch.
+	 */
+	const getNewSharedTreeBranchAndTaskGroup = (
+		sharedTreeAppState: SharedTreeAppState,
+		taskGroup: SharedTreeTaskGroup,
+	): {
+		currentBranch: TreeBranch & TreeViewAlpha<typeof SharedTreeAppState>;
+		newBranchTree: TreeBranch & TreeViewAlpha<typeof SharedTreeAppState>;
+		newBranchTaskGroup: SharedTreeTaskGroup;
+	} => {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const currentBranch = TreeAlpha.branch(sharedTreeAppState)! as TreeViewAlpha<
```

Comment:
As in the other file, we can use a ST API to make this safer instead of casting.

Reply:
Updated in commit XX